Lib/Schema/Table.php

```diff
@@ -0,0 +1,603 @@
+      if (!is_a($field, 'Freyja\Database\Schema\Field'))
+        throw InvArgExcp::typeMismatch('field', $field, 'Freyja\Database\Schema\Field');
+
+    $this->type = 'alter';
+    $this->alter_fields[$type] = array_merge($this->alter_fields, $fields);
+    return $this;
+  }
+
+  /**
+   * Build 'create' query string.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @return string
+   *
+   * @throws \RuntimeException if it is raised by the Field method called.
+   * @see Freyja\Database\Schema\Field::getField()
+   * @throws \LogicException if AUTO_INCREMENT is set on more than one field, or
+   * in a non primary key field, or if $fields property is an empty array.
+   */
+  private function buildCreate() {
+    $query = 'CREATE TABLE IF NOT EXISTS '.$this->name.' (';
+
+    $count = 0;
+    $autoinc_fields = 0;
+    if (empty($fields))
+      throw new LogicException('A table must have at least 1 column');
+    foreach ($this->fields as $field) {
+      if ($count != 0)
+        $query .= ', ';
+
+      try {
+        $field_info = $field->getField();
+      } catch (Exception $e) {
+        throw $e;
+      }
+      $field_name = key($field_info);
+      $info = $field_info[$field_name];
+
+      // Append field name and type.
+      $query .= $field_name.' '.$info['type'];
+
+      // Append default value, if set.
+      if (!is_null($info['default']))
+        $query .= ' DEFAULT '.$info['default'];
+
+      // Append NOT NULL, if true.
+      if ($info['NOT NULL'])
+        $query .= ' NOT NULL';
+
+      // Append UNSIGNED, if true.
+      if ($info['UNSIGNED'])
+        $query .= ' UNSIGNED';
+
+      // Append AUTO_INCREMENT, if true.
+      if ($info['AUTO_INCREMENT']) {
+        if ($autoinc_fields > 0)
+          throw new LogicException('AUTO_INCREMENT cannot be set on more than one field');
+        if (!in_array($info['name'], $this->primary_keys))
+          throw new LogicException('AUTO_INCREMENT cannot be set on a field that isn\'t primary key');
+
+        $query .= ' AUTO_INCREMENT';
+      }
+    }
+
+    // Append the PRIMARY KEY constraint, if set.
+    if (!empty($this->primary_keys)) {
+      $key_fields = '';
+      $count = 0;
+      foreach ($this->primary_keys as $key) {
+        if ($count != 0)
+          $key_fields .= ',';
+        $key_fields .= $key;
+      }
+      $query .= sprintf(
+        ', CONSTRAINT %1$s PRIMARY KEY (%2$s)',
+        $this->primary_name,
+        $key_fields
+      );
+    }
+
+    // Append the FOREIGN KEY constraints, if set.
+    foreach ($this->foreign_keys as $key) {
+      $query .= sprintf(
+        ', FOREIGN KEY (%1$s) REFERENCES %2$s(%3$s)',
+        $key['field'],
+        $key['references'],
+        $key['on']
+      );
+    }
+
+    // Append charset, collation and engine.
+    $query .= sprintf(
+      ') CHARACTER SET %1$s COLLATION %2$s ENGINE %3$s;',
+      $this->charset,
+      $this->collation,
+      $this->engine
+    );
```

Comment:
Yes, they are always set by the [constructor](https://github.com/SqueezyWeb/Database/pull/11/files#diff-6e11c83e48831bda794ee723c7db08ddR146).